src/objects/zcl_abapgit_objects.clas.abap

```diff
@@ -1033,11 +1034,19 @@ CLASS ZCL_ABAPGIT_OBJECTS IMPLEMENTATION.
 
     li_obj->serialize( li_xml ).
 
-    lo_files->add_xml( ii_xml      = li_xml
-                       is_metadata = li_obj->get_metadata( ) ).
-
     rs_files_and_item-files = lo_files->get_files( ).
 
+    LOOP AT rs_files_and_item-files TRANSPORTING NO FIELDS WHERE filename CP '*.json'.
+      lv_json_found = abap_true.  " means it can deal with json format
+      EXIT.
+    ENDLOOP.
+
+    IF lv_json_found = abap_false.
+      lo_files->add_xml( ii_xml      = li_xml
+                         is_metadata = li_obj->get_metadata( ) ).
+      rs_files_and_item-files = lo_files->get_files( ).
+    ENDIF.
+
```

Comment:
this wont work, if there is a W3MI object with a json file it will break

Reply:
Any better idea how to avoid redundant XML file for the objects supporting JSON format ? 
For CHKO, CHKV, CHKC the XML-file should not be added to rs_files_and_item-files...

Reply:
W3MI  has JSON and XML files? Sorry, have no expertise with this object type...

Reply:
Update see 75846a